### ISSUE TITLE
Adding a 'Buddy' make the app crash

### ISSUE THREAD

**Issue by pyknite** [ChatSecure/ChatSecure-iOS#388]
I created a new account on calyx server, logged on, tried to add a new buddy and the app crashed.

It seems ChatSecure can't import ChatSecureCore:

<img width="701" alt="screen shot 2015-09-23 at 17 25 51" src="https://cloud.githubusercontent.com/assets/203515/10049579/6e8423f6-6218-11e5-92a1-3cdcb439589a.png">

Here is the debugger log:

...
2015-09-23 17:25:04:125 ChatSecure[877:540b] RECV: <presence xmlns="jabber:client" from="tu@jabber.calyxinstitute.org/chatsecure90161"><x xmlns="vcard-temp:x:update"><photo/></x><c xmlns="http://jabber.org/protocol/caps" hash="sha-1" ver="aO0i9MaUBkx3mE7HcD49yGarihY=" node="https://github.com/robbiehanson/XMPPFramework"/></presence>
2015-09-23 17:25:04:125 ChatSecure[877:ac0f] OTRXMPPManager: xmppStream:didReceiveIQ: - F67A509F-18F0-4404-A4E8-3284C0AA8A2B
2015-09-23 17:25:04:153 ChatSecure[877:ac0f] OTRXMPPManager: xmppStream:didReceivePresence: - tu@jabber.calyxinstitute.org/chatsecure90161
Type: available
Show: (null)
Status: (null)
2015-09-23 17:25:11.322 ChatSecure[877:674805] *** Terminating app due to uncaught exception 'NSInvalidArgumentException', reason: '*** +[NSLayoutConstraint constraintWithItem:attribute:relatedBy:toItem:attribute:multiplier:constant:]: A constraint cannot be made between a leading/trailing attribute and a right/left attribute. Use leading/trailing for both or neither.'
*** First throw call stack:
(0x183198f5c 0x197d8bf80 0x183198ea4 0x1841ba30c 0x18403caf4 0x100eabd90 0x100eaaf74 0x100eaaec8 0x101bc85dc 0x188f19e8c 0x188f1a0f4 0x1887f483c 0x1887f3968 0x1887f4470 0x101bc843c 0x101bb1244 0x188a3167c 0x188a317d4 0x188a213d4 0x188a36364 0x1887d7a90 0x1886eb00c 0x187ef1f14 0x187eecb20 0x187eec9e0 0x187eec07c 0x187eebdd0 0x1886e1238 0x18314fc30 0x18314d9d4 0x18314de04 0x18307cdc0 0x18e1d0088 0x188756f60 0x100087ec8 0x1985b68b8)
libc++abi.dylib: terminating with uncaught exception of type NSException

I got ChatSecure to compile with Xcode 7 using issue #383 and issue #381. So I'm not sure if it's a ChatSecure problem or a problem coming from my 'hack'. Everything seems fine, but maybe something went wrong during linkage?

Cheers

**Comment by chrisballinger** [comment]
The develop branch is unstable right now and you'd be better off sticking to master or a tagged release for now. We'll be working on fixing these issues ASAP so stay tuned!

**Comment by chrisballinger** [comment]
If you can, please add the stack trace by adding a Obj-C breakpoint in Xcode and then pasting the results here

**Comment by pyknite** [comment]
Hum I have the same issue with the branch master.

I'll paste the stack trace later!

**Comment by pyknite** [comment]
Ok, now it fails when I try to create a new Jabber account, but it seems it the same bug/problem.
I'm on the master branch.
```
libc++abi.dylib: terminating with uncaught exception of type NSException
(lldb) bt
* thread #1: tid = 0xc5981, 0x00000001986d31e0 libsystem_kernel.dylib`__pthread_kill + 8, queue = 'com.apple.main-thread', stop reason = signal SIGABRT
    frame #0: 0x00000001986d31e0 libsystem_kernel.dylib`__pthread_kill + 8
    frame #1: 0x000000019879cf0c libsystem_pthread.dylib`pthread_kill + 112
    frame #2: 0x0000000198646b78 libsystem_c.dylib`abort + 140
    frame #3: 0x00000001974093f4 libc++abi.dylib`abort_message + 132
    frame #4: 0x0000000197425e98 libc++abi.dylib`default_terminate_handler() + 304
    frame #5: 0x0000000197d8c248 libobjc.A.dylib`_objc_terminate() + 124
    frame #6: 0x0000000197422f44 libc++abi.dylib`std::__terminate(void (*)()) + 16
    frame #7: 0x0000000197422b10 libc++abi.dylib`__cxa_rethrow + 144
    frame #8: 0x0000000197d8c120 libobjc.A.dylib`objc_exception_rethrow + 44
    frame #9: 0x000000018307ce68 CoreFoundation`CFRunLoopRunSpecific + 552
    frame #10: 0x000000018e1d0088 GraphicsServices`GSEventRunModal + 180
    frame #11: 0x0000000188756f60 UIKit`UIApplicationMain + 204
  * frame #12: 0x00000001000b89d0 ChatSecure`main(argc=1, argv=0x000000016fdd7af0) + 124 at main.m:14
    frame #13: 0x00000001985b68b8 libdyld.dylib`start + 4
(lldb) 
```
Hope it helps

**Comment by pyknite** [comment]
Just tried with Xcode 6.4, works fine!

**Comment by chrisballinger** [comment]
Haha if only the stack trace included the offending UI constraints :(

**Comment by pyknite** [comment]
Sorry, I would have dug deeper, but I don't have the time right now :( 

I forgot to say that in Xcode 6.4 I tried with the default compiler for 6.4 (Apple clang-600) and the last release of LLVM (the OSS version, v3.7) and both worked.

**Comment by chrisballinger** [comment]
Fixed in ae94a2e34dc54c632b406d3780a25fe923cb9fd2
